Add data-prevent-double-click to submit buttons

chubberlisk opened the pull request:
### Context
While updating to version 3 of `govuk-frontend` we found that forms could be double submitted which would add excess data to the database. The GOV.UK frontend has the ability to handle this (see https://design-system.service.gov.uk/components/button/#stop-users-from-accidentally-sending-information-more-than-once).

### Changes proposed in this pull request

`'data-prevent-double-click': true` has been added to the Personal Details, Contact Details and Degrees forms.

### Guidance to review

You can check this by using the Developer tools and changing the network speed to "Slow 3G" and attempting to double submit with and without the `data-prevent-double-click` present.

### Link to Trello card

[691 - Prevent double-click form submission](https://trello.com/c/9GB7aXwJ/691-prevent-double-click-form-submission)

duncanjbrown commented:
It looks like you can pass this as an attribute to the `f.submit` method? https://github.com/DFE-Digital/govuk_design_system_formbuilder/blob/9806a0bc36ca1de13a655d74c629ba7b2c217039/lib/govuk_design_system_formbuilder/elements/submit.rb#L9

alexdesi commented on app/views/contact_details/_form.html.erb:
```diff
@@ -13,7 +13,7 @@
       <%= f.govuk_text_area :address, label: { text: t('application_form.contact_details_section.address.label')} %>
 
-      <%= f.submit t('application_form.save_and_continue'), class: 'govuk-button' %>
+      <%= f.submit t('application_form.save_and_continue'), class: 'govuk-button', 'data-prevent-double-click': true %>
```
magic 👍

chubberlisk commented:
> It looks like you can pass this as an attribute to the `f.submit` method? https://github.com/DFE-Digital/govuk_design_system_formbuilder/blob/9806a0bc36ca1de13a655d74c629ba7b2c217039/lib/govuk_design_system_formbuilder/elements/submit.rb#L9

Yup! I've updated the button to be a `f.govuk_submit`button and added `prevent_double_click`.

chubberlisk commented:
> > It looks like you can pass this as an attribute to the `f.submit` method? https://github.com/DFE-Digital/govuk_design_system_formbuilder/blob/9806a0bc36ca1de13a655d74c629ba7b2c217039/lib/govuk_design_system_formbuilder/elements/submit.rb#L9
> 
> Yup, thanks! I've updated the button to be a `f.govuk_submit` button and added `prevent_double_click` attribute now.

We've decided to revert back to `#submit` to allow for the `data-module` attribute to be added because `#govuk_submit` does not support it yet as it is a very recent addition to the GOV.UK frontend.
